### PR TITLE
Remove old obsolete validators and revamp validator schema

### PR DESCRIPTION
**Pull request by GeeWee**
This PR removes the old pandas-based validators in places where we can efficiently cover them with the new column-based validators.

It also documents the validators better and revamps a few validators based on discussions in the OEFDB repository, most notably:
- Changing the `allow_empty` logic to always determine how to handle empty strings. Empty strings are no longer passed to the validators if `allow_empty` is set to `false` - it is simply discarded. 
- Changing `is_allowed_string` to `has_no_commas`
- Added `is_legal_id` so we can deprecate the old id checker.

### CODE REVIEW

**Comment by GeeWee** [comment]
@AnastasiaLobanova would love your take on the validator naming
@grindcode feel free to give the code a lookaroo

File: README.md [hunk 1]
```diff
@@ -64,6 +64,37 @@ oefdb_df_under_review = oefdb.import_from_github(pr=123)
 2. Use Python to review the data, comparing it to the main revision data as necessary.
 3. Add review comments via the GitHub UI as necessary and approve when discussions are resolved.
 
+## Schema validation
```
Comment (grindcode) [inline README.md:67]: Thanks!

File: oefdb/validators/schema/cell_validators.py [hunk 2]
```diff
@@ -23,20 +24,29 @@ def validate_cell(self, cell_value: str) -> cell_validator_return_type:
...
-            return None
+        return None
 
     return f"'{cell_value}' was not a valid string"
 
 
+valid_id_punctuation_regex = re.compile(r"-|_|\.")
+
+
+def is_legal_id(cell_value: str) -> cell_validator_return_type:
```
Comment (grindcode) [inline oefdb/validators/schema/cell_validators.py:41]: For consistent terminology, why not `is_valid_id`?

**Review by grindcode — APPROVED** [review]
Lgtm

File: oefdb/validators/schema/cell_validators.py (hunk 2 above)
Comment (GeeWee) [reply oefdb/validators/schema/cell_validators.py:41]: Yes absolutely

**Pull request merged**